sdk/identity/Azure.Identity/src/TokenCredentialProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    public class TokenCredentialProvider : TokenCredential
+    {
+        private TokenCredential[] _sources;
+
+        public TokenCredentialProvider(params TokenCredential[] sources)
+        {
+            if (sources == null) throw new ArgumentNullException(nameof(sources));
+
+            if (sources.Length == 0) throw new ArgumentException("sources must not be empty", nameof(sources));
+
+            if (sources.Any(tc => tc == null)) throw new ArgumentException("sources must not contain null", nameof(sources));
```

Comment:
I don't think we should be using linq in libraries. It's easy to result in unwanted allocations.